### DISCUSSION TITLE
External DTDs

### DISCUSSION THREAD

**Discussion by Devasta** [ballsteve/xrust#32] [Planning]
Hi Steve,

External DTDs, just wondering if it's ok that I put a restriction on the processor that the user must provide a pre-approved list of URLs for the DTDs, possibly as command line args or in some config file? Figure it'd be a bit of a security risk otherwise.

I'll probably also create a folder somewhere that'll hold popular DTDs like XHTML.

Let me know what you think.

**Comment by ballsteve** [comment]
There are a few places where security is required. The code needs to support 'security policies' that the user can provide or override.

Of course it's OK to put in the restriction. Do something simple to get started, and we can develop a more sophisticated mechanism at some point in the future.

**Reply by ballsteve** [reply]
Actually, it goes a deeper than security concerns.

One subproject I want to work on is getting χrust working with wasm. The last time I tried to get it up-and-running it failed miserably :-(  I finally worked out that the failure was due to the xslt module pulling in networking code, in order to support loading external resources (via xsl:import and xsl:include).

My current thinking is that we will need to separate code that requires networking from the main code. This will probably need to be done using closures. This will hopefully dovetail with the security policy mechanism.

**Reply by Devasta** [reply]
So it would need to work on same domain if run in WASM, but we'd want a set list of directories, probably only the directory the Stylesheet was located in, for running locally?

Definitely right, something that is going to come about from whatever solution is used for networking.
